src/api/firebase.js

```diff
@@ -71,12 +73,24 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listId, itemId) {
+	const listItemRef = doc(db, listId, itemId);
+	const listItemSnap = await getDoc(listItemRef);
+	const totalPurchases = listItemSnap.data().totalPurchases;
+	await updateDoc(listItemRef, {
+		dateLastPurchased: new Date(),
+		totalPurchases: totalPurchases + 1,
+	});
+}
+
+export async function undoUpdateItem(listId, itemId) {
+	const listItemRef = doc(db, listId, itemId);
+	const listItemSnap = await getDoc(listItemRef);
+	const totalPurchases = listItemSnap.data().totalPurchases;
+	await updateDoc(listItemRef, {
+		dateLastPurchased: null,
+		totalPurchases: totalPurchases - 1,
+	});
 }
```

Comment:
I really like that you added the ability to uncheck an item! I'm wondering if this could be made into one function `updateItem` with a new parameter of `checked` to decide the values of `dateLastPurchased` and `dateLastPurchased`.  

Then you can even add a nice comment explaining what `checked` is doing!

Reply:
Love this feedback, great way to be DRY!